### PR TITLE
MultiSlider refactors

### PR DESCRIPTION
**Pull request by giladgray**
see individual commits.

- rename member methods to be more obvious about purpose.
- set border-radius only on track (never on segments)

### CODE REVIEW

File: packages/core/src/components/slider/multiSlider.tsx [hunk 1]
```diff
@@ -198,27 +201,23 @@ export class MultiSlider extends AbstractPureComponent<IMultiSliderProps, ISlide
...
             return value.toFixed(this.state.labelPrecision);
         }
     }
 
-    private maybeRenderAxis() {
-        // explicit typedefs are required because tsc (rightly) assumes that props might be overriden with different
-        // types in subclasses
-        const max: number = this.props.max;
-        const min: number = this.props.min;
-        const labelStepSize: number = this.props.labelStepSize;
```
Comment (giladgray) [inline packages/core/src/components/slider/multiSlider.tsx]: this is just not necessary anymore (no more subclass)

File: packages/core/src/components/slider/multiSlider.tsx [hunk 2]
```diff
@@ -198,27 +201,23 @@ export class MultiSlider extends AbstractPureComponent<IMultiSliderProps, ISlide
     private formatLabel(value: number): React.ReactChild {
         const { labelRenderer } = this.props;
         if (labelRenderer === false) {
-            return undefined;
+            return null;
         } else if (Utils.isFunction(labelRenderer)) {
-            // TODO: TS 2.7 might have a type narrowing issue?
-            return (labelRenderer as (value: number) => React.ReactChild)(value);
```
Comment (giladgray) [inline packages/core/src/components/slider/multiSlider.tsx]: resolved now (maybe due to removing subclass)

File: packages/core/src/components/slider/multiSlider.tsx [hunk 3]
```diff
@@ -228,78 +227,42 @@ export class MultiSlider extends AbstractPureComponent<IMultiSliderProps, ISlide
...
-                this.props.defaultTrackIntent,
-            ];
-            const fillIntent = fillIntentPriorities.filter(intent => intent != null)[0];
-            tracks.push(this.renderTrackFill(index, left, right, fillIntent));
+    private renderTracks() {
+        const trackStops = getSortedHandleProps(this.props);
+        trackStops.push({ value: this.props.max });
+
+        // render from current to previous, then increment previous
+        let previous: ISliderHandleProps = { value: this.props.min };
```
Comment (jscheiny) [inline packages/core/src/components/slider/multiSlider.tsx]: Does this still correctly handle non-interactive handles outside the slider range? Specifically, we had what we had previously to deal with cases like this:

```typescript
<Slider min={5} max={10} value={7} showTrackFill={true} />
```

Comment (giladgray) [reply packages/core/src/components/slider/multiSlider.tsx]: the `clamp()` in getOffsetRatio ensures segments must be between 0 and 1.

Comment (giladgray) [reply packages/core/src/components/slider/multiSlider.tsx]: added test case to be usre

**Pull request merged**